File: java/205. Isomorphic Strings.java
public class Solution {
    public boolean isIsomorphic(String s, String t) {
        if (s.length()!=t.length()) {
			return false;
		}
        Map<String,Integer> sMap=new HashMap<String, Integer>();
        Map<String,Integer> tMap=new HashMap<String, Integer>();
        for (int i = 0; i < s.length(); i++) {
			String sk=s.substring(i,i+1);
			String tk=t.substring(i,i+1);
			if ((sMap.get(sk)==null)!=((tMap.get(tk)==null))) {
				return false;
			}else if (sMap.get(sk)==null) {
				sMap.put(sk, i);
				tMap.put(tk, i);
			}else if(!sMap.get(sk).equals(tMap.get(tk))){
				return false;
			}else {
				sMap.put(sk, i);
				tMap.put(tk, i);
			}
		}
        return true;
    }
}

File: python/205. Isomorphic Strings.py
#coding=utf-8
__author__ = 'xuxuan'
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s)!=len(t):
            return False
        smap={}
        tmap={}
        for i in range(len(s)):
            if smap.get(s[i]) or tmap.get(t[i]):
                if smap.get(s[i])!=t[i] or tmap.get(t[i])!=s[i]:
                    return False
            smap[s[i]]=t[i]
            tmap[t[i]]=s[i]
        return True